generate_ci.py

```diff
@@ -0,0 +1,416 @@
+
+            print(f"=== {name} ===")
+            checkout_branch(name, supported_branch)
+
+            test_targets = get_test_targets(name, r)
+            print("---> Matching targets", test_targets)
+
+            build_image = "openshift/ci-operator/build-image"
+            print("---> Build image", build_image)
+
+            images = get_images(path="openshift/ci-operator/knative-images", name=name, r=r)
+            print("---> Images", images)
+            test_images = get_images(path="openshift/ci-operator/knative-test-images", name=name, r=r,
+                                     prefix_ctx="test-")
+            print("---> Test images", test_images)
+
+            mirroring_lines = []
+            for img in test_images:
+                frm = f"{openshift_release_registry}/{promotion_name}:{img.name}"
+                to = f"{quay_registry}/{img.name}:{promotion_name}"
+                mirroring_lines.append(f"{frm} {to}\n")
+
+            file_name = f"mapping_{promotion_name}_{repo_name}_quay"
+            with open(f"{openshift_release_image_mirroring}/{file_name}", "w") as f:
+                f.writelines(mirroring_lines)
+
+            images_field = ""
+            dependencies = ""
+            for img in images + test_images:
+                images_field += f"""
+- dockerfile_path: openshift/{img.path}/Dockerfile
+  to: {img.name}"""
+
+                dependencies += f"""
+      - env: {img.env_name}
+        name: {img.name}"""
+
+            tests_field = ""
+            continuous = ""
+            schedule = ""
+
+            for x in range(2):
+                for test_target in test_targets:
+                    tests_field += f"""
+- as: {test_target}-aws-ocp-{ov.replace('.', '')}{continuous}
+  cluster_claim:
+    architecture: amd64
+    cloud: aws
+    owner: openshift-ci
+    product: ocp
+    timeout: 1h0m0s
+    version: "{ov}"
+  {schedule}
+  steps:
+    test:
+    - as: test
+      cli: latest
+      commands: make {test_target}
+      dependencies: {dependencies}
+      from: src
+      resources:
+        requests:
+          cpu: 100m
+      timeout: 4h0m0s
+    workflow: generic-claim
+            """
+                continuous = "-continuous"
+                schedule = "cron: 0 5 * * 2,6"
+
+            ci = f""" # Generated by generate_ci.py
+base_images:
+  base:
+    name: "{ov}"
+    namespace: ocp
+    tag: base
+build_root:
+  project_image:
+    dockerfile_path: {build_image}/Dockerfile
+images: {images_field}
+promotion:
+  additional_images:
+    {name[name.index("/") + 1:]}-src: src
+  disabled: {"true" if openshift_versions[-1] != ov else "false"}
+  name: {promotion_name}
+  namespace: {openshift_release_registry_namespace}
+releases:
+  initial:
+    integration:
+      name: "{ov}"
+      namespace: ocp
+  latest:
+    integration:
+      include_built_images: true
+      name: "{ov}"
+      namespace: ocp
```

Comment:
AFAIK this whole part doesn't have any effect since we started to rely on `cluster_claim` pools. Those should control and define OCP version etc.

Reply:
Thanks!

Reply:
which part exactly?

Reply:
the entire?
```
releases:
  initial:
    integration:
      name: "{ov}"
      namespace: ocp
  latest:
    integration:
      include_built_images: true
      name: "{ov}"
      namespace: ocp
```

Reply:
Yes, whole block is L341-350.